File: SagaPlay-Backend/CatalogService/Models/ContentItem.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace CatalogService.Models
{
    
    public class ContentItem
    {
        public int Id { get; set; }

        public string Title { get; set; }

        public string PlotSummary { get; set; }

        public DateTime ReleaseDate { get; set; }

        public string Genre { get; set; }

        public string Director { get; set; }        

        public string Rating { get; set; }

        public string PosterURL { get; set; }

        //Navigation properties       
        public ICollection<CastMember> MainCast { get; set; }

    }
}

File: SagaPlay-Frontend/src/app/features/catalog/contentitem.ts
export interface ContentItem{
    Id: number;    
    Title: string;
    PlotSummary: string;
    ReleaseDate: string;
    Genre: string;
    Director: string;
    Rating: string;
    PosterURL: string;
    MainCast: CastMemberDTO[];
}

interface CastMemberDTO{    
    Id:number;
    Name:string;
    Gender:string;
}